### ISSUE TITLE
Could we not use ObjectBox for storing data? 

### ISSUE THREAD

**Issue by 2hamed** [persiancal/android-sdk#15]
Since this is a SDK and it's better for libraries and SDKs to not depend on third-party package I suggest we move away from ObjectBox to native Sqlite for storing data.

**Comment by bkhezry** [comment]
@2hamed with using native SQLite we should provide too many codes only to handle CRUD. ObjectBox is so tiny library and useful. In the end, for each solution the `.so` files are inevitable.

**Comment by 2hamed** [comment]
But sqlite is provided by the OS and no additional `.so` file is needed. And who knows, it may lead to dependency conflicts if the users are using ObjectBox themselves.

**Comment by bkhezry** [comment]
@2hamed it's correct. It may be conflicts in creating Singleton instance of `BoxStore` class. I will be try to prevent from it. As you know, I use some library like retrofit & gson too in this sdk. We can't replace all of them.

**Comment by 2hamed** [comment]
Yes, I understand and I may be able to help developing it further but to do so I need a clear vision of this library is supposed to do and what features are planned. Is there any way we can connect?

**Comment by bkhezry** [comment]
@2hamed bkhezry at Gmail
